pkg/util/podreadiness/pod_readiness_utils.go

```diff
@@ -61,8 +68,13 @@ func AddNotReadyKey(c client.Client, pod *v1.Pod, msg Message) error {
 			condition.LastTransitionTime = metav1.Now()
 		}
 
-		return c.Status().Update(context.TODO(), newPod)
+		err := c.Status().Update(context.TODO(), newPod)
```

Comment:
The old logic here can be optimized that we'd better update the Ready condition to False after set KruisePodReady condition to False. Like what we do in inplace update:

https://github.com/openkruise/kruise/blob/5b52b6eff3ff93210206eeaeb79751a16043bfd9/pkg/util/inplaceupdate/inplace_update.go#L172-L175